Kernel_23/include/CGAL/Kernel/function_objects.h

```diff
@@ -3215,18 +3215,55 @@ namespace CommonKernelFunctors {
     }
 
     typename K::Point_3
-    operator()(const typename K::Ray_3& ray,
-               const typename K::Point_3& query,
-               const K& k)
+      operator()(const typename K::Ray_3& ray,
+        const typename K::Point_3& query,
+        const K& k)
     {
-      if ( ray.to_vector() * (query-ray.source()) <= 0)
+      if (ray.to_vector() * (query - ray.source()) <= 0)
```

Comment:
unrelated to this PR but this is not "robust" with EPICK. The safest solution would be to always build the projection onto the line and then use that point for the orientation test to either return the source or the constructed point.